R/settings.R

```diff
@@ -50,18 +39,45 @@ read_settings <- function(filename) {
   }
 }
 
+read_config_file <- function(config_file) {
+  if (is.null(config_file)) {
+    return(NULL)
+  }
+
+  if (endsWith(config_file, ".R")) {
+    config <- new.env()
+    read <- function(file) {
+      sys.source(file, config)
```

Comment:
How safe is this with respect to file encodings?

Reply:
Not sure... but `read.dcf()` would have the same issue, right?

Reply:
it's using `readLines()`, so in principle we could add encoding to the a `conn` object passed in...

https://github.com/r-devel/r-svn/blob/d12e38f0c510e3315265e7ada26e3e33370278ba/src/library/base/R/source.R#L269

Similarly, `read.dcf()` accepts a connection, could do the same thing.

Reply:
Maybe worth assuming the file is UTF-8?
But fine as a follow up if the current behavior is locale-dependent as well.

Reply:
#2181